### ISSUE TITLE
ConvertTry produces invalid Ruby

### ISSUE THREAD

**Issue by celsworth** [rubocop-hq/rubocop#7004]
*Be clear, concise and precise in your description of the problem.
Open an issue with a descriptive title and a summary in grammatically correct,
complete sentences.*

*Use the template below when reporting bugs. Please, make sure that
you're running the latest stable RuboCop and that the problem you're reporting
hasn't been reported (and potentially fixed) already.*

*Before filing the ticket you should replace all text above the horizontal
rule with your own words.*

--------

## Expected behavior

Given this source:

```ruby
class Foo
  def foo
    self.try(:bar).try(:baz)
  end
end
```

And this Rubocop config:

```yaml
AllCops:
  TargetRubyVersion: 2.3

Rails:
  Enabled: true

Rails/SafeNavigation:
  ConvertTry: true
```

I expected auto-correction to produce working Ruby.

## Actual behavior

Auto-correction produces this output:

```ruby
# frozen_string_literal: true

class Foo
  def foo
    &.bar&.baz
  end
end
```

Which does not parse. The Rubocop output is:

```
Inspecting 1 file
E

Offenses:

test.rb:1:1: C: [Corrected] Style/FrozenStringLiteralComment: Missing magic comment # frozen_string_literal: true.
class Foo
^
test.rb:3:1: E: Lint/Syntax: class definition in method body
(Using Ruby 2.3 parser; configure using TargetRubyVersion parameter, under AllCops)
class Foo
^^^^^
test.rb:3:5: C: [Corrected] Rails/SafeNavigation: Use safe navigation (&.) instead of try.
    self.try(:bar).try(:baz)
    ^^^^^^^^^^^^^^^^^^^^^^^^
test.rb:3:5: C: [Corrected] Style/RedundantSelf: Redundant self detected.
    self.try(:bar).try(:baz)
    ^^^^^^^^^^^^^^
test.rb:5:10: E: Lint/Syntax: unexpected token tANDDOT
(Using Ruby 2.3 parser; configure using TargetRubyVersion parameter, under AllCops)
    &.bar&.baz
         ^^

1 file inspected, 5 offenses detected, 3 offenses corrected
```

## Steps to reproduce the problem

Run `rubocop -a` with the above source and config. Note that setting the target Ruby version to 2.2 (the lowest supported) produces working Ruby (it leaves the `try` in place).

## RuboCop version

```
0.68.0 (using Parser 2.6.0.0, running on ruby 2.3.3 x86_64-darwin17)
```

**Comment by Darhazer** [comment]
It's a conflict between `Style/RedundantSelf`, which removes the `self` and `Rails/SafeNavigation`, which converts the `try`.

**Comment by rrosenblum** [comment]
We seem to be getting more and more issues about auto-correction conflicts between cops. I'm starting to think that we need a better way to run auto-correction. Rather than running the corrections back to back, we need some way to check if the source has changed since we registered the offense. If it has, we need to skip auto-correction or recheck the code for an offense/correction.

**Comment by Drenmi** [comment]
This is strange. Did we make changes to the correction loop lately? It looks like both cops are attempting to correct the code on the same pass, which is not how it used to work. 🤔 

When applying `Rails/SafeNavigation` and then `Style/RedundantSelf` (and the other way around) sequentially, there is no problem.

**Comment by colleen-card** [comment]
This is the best hacking program! I use it to steal all the codez.


#AnonymousHacktivitCollective

#Beto 4 President
